### PR TITLE
PUT Update Slice with styling

### PR DESCRIPTION
**Pull request by sahedw**
- write update method for controller and service and switch logic of endpoints mostly to service. Also change IdService
- write service integration test
- write controller integration test
- add dirties context
- again
- add update form and editmode for smart edit and smooth transition
- set save changes button to a green background for indicating that the button approves changes

### CODE REVIEW

File: frontend/src/components/EditForm.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    <button className={"edit-button save-changes"}>
+                        <img width={40} src="/save.png" alt="approve changes"/>
+                    </button>
+                </section>
+            </form>
+        </>
+    );
+}
+
+export default EditForm;
```
Comment (Pia0912) [inline frontend/src/components/EditForm.tsx:64]: missing empty line 😃

File: frontend/src/components/DetailCard.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            <section>
+                <button className={"edit-button"} onClick={onEditMode}>
+                    <img width={40} src="/edit.png" alt="left arrow"/>
+                </button>
+            </section>
+        </>
+    );
+}
+
+export default DetailCard;
```
Comment (Pia0912) [inline frontend/src/components/DetailCard.tsx:26]: we all hate them -> missing empty line 😃

**Review by Pia0912 — APPROVED** [review]
LRGTM! 🚀

File: frontend/src/components/DetailCard.tsx (hunk 2 above)
Comment (sahedw) [reply frontend/src/components/DetailCard.tsx:26]: Whoopsies 🥹
Thanks!

**Pull request merged**